pkg/zfs/mount.go

```diff
@@ -60,29 +61,28 @@ func UmountVolume(vol *apis.ZFSVolume, targetPath string,
 		return nil
 	}
 
-	if vol.Spec.VolumeType == VOLTYPE_DATASET {
-		if err = UmountZFSDataset(vol); err != nil {
-			logrus.Errorf(
-				"zfspv failed to umount dataset: path %s Error: %v",
-				targetPath, err,
-			)
-			return err
-		}
-	} else {
-		if err = mounter.Unmount(targetPath); err != nil {
-			logrus.Errorf(
-				"zfspv failed to unmount zvol: path %s Error: %v",
-				targetPath, err,
-			)
-			return err
-		}
+	if err = mounter.Unmount(targetPath); err != nil {
```

Comment:
q_: Is the case of volume (with dataset), invoking unmount prior to upgrade handled by this case?

Reply:
> invoking unmount prior to upgrade
did not get the question, Prior to upgrade, no need to worry, as old code will handle the umount request via zfs umount command.

Here since umount works on dataset as well as zvol. So having one code to handle both the cases.